net/icmpv6/icmpv6_input.c

```diff
@@ -336,10 +336,12 @@ void icmpv6_input(FAR struct net_driver_s *dev, unsigned int iplen)
 
                     if ((prefixopt->flags & ICMPv6_PRFX_FLAG_A) != 0)
                       {
+                        FAR void *addr = prefixopt->prefix;
```

Comment:
But this will not make the pointer aligned.

Reply:
I have no ideas... just trick the compiler, do you have any suggestions?

Reply:
You could copy `prefix` out of the packed structure. The target array would then be properly aligned.

Reply:
If the pointer may indeed be unaligned, then the only option here is to make a tmp net_ipv6addr_t variable and copy prefix. Then, tmp is passed to function call.

Reply:
Or introduce intermediate type for ICMPv6_OPT_PREFIX that will be not packed

Reply:
emm.... It seems that packed struct is not a good solution, there are still many warnings of `Werror=address-of-packed-member`, If the `prefix` needs to be copied, I think which need to consider other ways to fix this issue, Let me revert related changes, thanks for the advice!  @gustavonihei @pkarashchenko